### PR TITLE
Add the version of find_backward() that is actaully going in to the s…

### PR DESCRIPTION
**Pull request by tzlaine**
…tandard -- find_last().

### CODE REVIEW

File: include/boost/algorithm/find_last.hpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        typename Proj = identity,
+        std::indirect_unary_predicate<std::projected<iterator_t<R>, Proj>> Pred>
+#else
+    template<typename R, typename Pred, typename Proj = identity>
+#endif
+    BOOST_CXX14_CONSTEXPR subrange<iterator_t<R>>
+    find_last_if_not(R && r, Pred pred, Proj proj = {})
+    {
+        return algorithm::find_last_if_not(
+            std::begin(r), std::end(r), pred, proj);
```
Comment (glenfe) [inline include/boost/algorithm/find_last.hpp:201]: Shouldn't this call begin and end unqualified? 

```
using std::begin;
using std::end;
return algorithm::find_last_if_not(begin(r), end(r), pred, proj);
```

File: include/boost/algorithm/find_last.hpp [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    requires std::indirect_binary_predicate < std::ranges::equal_to,
+        std::projected<iterator_t<R>, Proj>,
+    const T * >
+#else
+    template<typename R, typename T, typename Proj = identity>
+#endif
+        BOOST_CXX14_CONSTEXPR subrange<iterator_t<R>>
+        find_last(R && r, const T & value, Proj proj = {})
+    {
+        return algorithm::find_last(std::begin(r), std::end(r), value, proj);
```
Comment (glenfe) [inline include/boost/algorithm/find_last.hpp:120]: Same question here regarding ADL for `begin` and `end`.

File: include/boost/algorithm/cxx20_helper.hpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    struct subrange
+        : stl_interfaces::view_interface<subrange<Iterator, Sentinel>>
+    {
+        BOOST_CXX14_CONSTEXPR subrange() = default;
+        BOOST_CXX14_CONSTEXPR subrange(Iterator first, Sentinel last) :
+            first_(first), last_(last)
+        {}
+        template<typename R>
+        BOOST_CXX14_CONSTEXPR explicit subrange(const R & r) :
+            first_(std::begin(r)), last_(std::end(r))
```
Comment (glenfe) [inline include/boost/algorithm/cxx20_helper.hpp:51]: Same question here about ADL for `begin` and `end`

**Comment by tzlaine** [comment]
Good call.  I'll make another PR with the fix.

**Pull request closed without merging**